### ISSUE TITLE
MATLAB bindings on Linux may be troubled by differing libstdc++ versions.

### ISSUE THREAD

**Issue by chrisdembia** [opensim-org/opensim-core#1397]
Our MATLAB bindings rely on `libstdc++.so`. MATLAB itself also needs `libstdc++.so` and tries to use the one that is packaged with MATLAB instead of the one on the system. The two can often be inconsistent, and this lead to mysterious errors when trying to use the MATLAB bindings. There are at least two fixes, neither of which are great:

1. The user can set the `LD_PRELOAD` environment variable before starting MATLAB to force MATLAB to load the correct libstdc++.so:
  ```
  LD_PRELOAD="/usr/lib/x86_64-linux-gnu/libstdc++.so.6" matlab
  ```

2. Overwrite MATLAB's libstdc++.so, which is located at a path like `/usr/local/MATLAB/R2016b/sys/os/glnx64/libstdc++.so.6`, with the one from the system.

We could also try to improve the error message that users get, so that we can tell them about the two possible fixes above.

  For more information, see [this SimTK forum post](https://simtk.org/plugins/phpBB/viewtopicPhpbb.php?f=91&t=7165&p=0&start=0&view=) and see [this MATLAB forum post with solutions to the problem](https://www.mathworks.com/matlabcentral/newsreader/view_thread/255846).

**Comment by chrisdembia** [comment]
Other solutions:

1. Using RPATH for `libosimCommon`, etc. should make sure the bindings find the correct libstdc++.

2. Statically link to libstdc++.

For more information about solutions like this, see http://stackoverflow.com/questions/13636513/linking-libstdc-statically-any-gotchas

**Comment by chrisdembia** [comment]
I think the best solution is using RPATH, as @klshrinidhi is achieving with #1354 .

**Comment by chrisdembia** [comment]
@klshrinidhi suggested seeing if this was now fixed, since #1354 is merged. I tested locally, and this issue does still exist. The directory containing libstdc++ does not get added to the RPATH for OpenSim's libraries; only the OpenSim install lib dir and Simbody's install lib dir were added.
